src/wallet/wallet.h

```diff
@@ -171,6 +171,15 @@ class CWallet : public CCryptoKeyStore
     }
 
     std::map<uint256, CWalletTx> mapWallet GUARDED_BY(cs_wallet);
+
+    //!
+    //! \brief This map is a cross reference from the transaction id (hash)
+    //! to the block index pointer. This pointer must also be in mapBlockIndex.
+    //! This map is used to cache txs in the wallet that are already in blocks
+    //! to avoid the expense of looking this up each time that ResendWalletTransactions
+    //! runs, for example.
+    std::map<uint256, CBlockIndex*> mapWalletTxBlockIndexRef GUARDED_BY(cs_wallet);
```

Comment:
Will it work with `unordered_map` here?

Reply:
Yes. Probably a good idea since the traversal is on mapWallet and not mapWalletTxBlockIndexRef, where single finds are done.

Reply:
Quite frankly when I really analyzed this ResendWalletTransactions, I couldn't believe how crazy this code is. Even worse... why doesn't the CWallet transaction serialized to the disk in wallet.dat not contain a field to indicate that it was already included in a block?

Reply:
Wait, I think CWalletTx derives from CMerkleTx, which has a hashBlock.

Reply:
Hmm....
CXX      rpc/libgridcoin_util_a-server.o
  CXX      rpc/libgridcoin_util_a-voting.o
  CXX      libgridcoin_util_a-validation.o
  CXX      wallet/libgridcoin_util_a-rpcdump.o
  CXX      wallet/libgridcoin_util_a-rpcwallet.o
  CXX      wallet/libgridcoin_util_a-wallet.o
  CXX      wallet/libgridcoin_util_a-walletdb.o
In file included from ./init.h:9,
                 from gridcoin/backup.cpp:6:
./wallet/wallet.h: In constructor ‘CWallet::CWallet()’:
./wallet/wallet.h:113:5: error: use of deleted function ‘std::unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>::unordered_map() [with _Key = uint256; _Tp = CBlockIndex*; _Hash = std::hash<uint256>; _Pred = std::equal_to<uint256>; _Alloc = std::allocator<std::pair<const uint256, CBlockIndex*> >]’
  113 |     {
      |     ^
In file included from /usr/include/c++/11/unordered_map:47,
                 from /usr/include/c++/11/functional:61,
                 from /usr/include/boost/system/error_code.hpp:22,
                 from /usr/include/boost/filesystem/path_traits.hpp:23,
                 from /usr/include/boost/filesystem/path.hpp:25,
                 from /usr/include/boost/filesystem.hpp:16,
                 from ./fs.h:14,
                 from ./gridcoin/backup.h:8,
                 from gridcoin/backup.cpp:5:
/usr/include/c++/11/bits/unordered_map.h:141:7: note: ‘std::unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>::unordered_map() [with _Key = uint256; _Tp = CBlockIndex*; _Hash = std::hash<uint256>; _Pred = std::equal_to<uint256>; _Alloc = std::allocator<std::pair<const uint256, CBlockIndex*> >]’ is implicitly deleted because the default definition would be ill-formed:
  141 |       unordered_map() = default;
      |       ^~~~~~~~~~~~~

Reply:
If the hashBlock field works correctly in the CWallet object, then we may be able to get rid of the cache map and use the hashBlock field as the sole determinant.

Reply:
I completely rewrote the approach and eliminated the cache map.